docs/source/agent-framework/web-framework.rst

```diff
@@ -14,29 +14,76 @@ Web SubSystem
 
-Methods
+.. code-block:: python
+
+    class WebAgent(Agent):
+        def __init__(self, **kwargs):
+            super(WebAgent, self).__init__(enable_web=True,**kwargs)
+
+
+MANIFEST File
+-------------
+
+The MANIFEST.in file is used to package additional files needed for your web enabled agent.
+Please read the python packaging `documentation <https://packaging.python.org/guides/using-manifest-in/>`_
+on the MANIFEST.in file. For most cases, i.e. when you only need to include a webroot directory for html
+and javascript, the manifest file only needs to include the `recursive-include` command. For example, the entirety
+of the VolttronCentral MANIFEST.in file is:
+
+.. code-block:: python
+
+    recursive-include volttroncentral/webroot *
+
+The MANIFEST.in file should be located in the root directory of the agent. All pathing for the MANIFEST.in file
+commands are relative to this root directory.
+
+Routes
 -------
 
-The web subsystem allows an agent to register three different types of endpoints; path based, JSON and websocket.  A path based endpoint allows the agent to specify a prefix and a static path on the file system to serve static files.  The prefix can be a regular expression.
+The web subsystem allows an agent to register three different types of routes; file paths, endpoints, and websockets.
 
-.. note:: The web subsystem is only available when the constructor contains enable_web=True.
+.. note::
+    For all routes the first match wins.  Therefore ordering which routes are registered first becomes important.
 
-The below examples are within the context of an object that has extended the :class:`volttron.platform.vip.agent.Agent` base class.
 
-.. note:: For all endpoint methods the first match wins.  Therefore ordering which endpoints are registered first becomes important.
+File Path
+~~~~~~~~~
+
+A path-based route that allows the agent to specify a prefix and a static path on the file system to serve static files.
+The prefix can be a regular expression.
+
+.. note::
+    The static path should point to a location within the installed agent's agent-data directory.
+    You MUST have read access to the directory.
+
+The below example is based on the registered static route in the SimpleWebAgent.
+
 
 .. code-block:: python
 
     @Core.receiver('onstart')
     def onstart(self, sender, **kwargs):
         """
-        Allow serving of static content from /var/www
+        Allow serving of static content from 'webroot'
         """
-        self.vip.web.register_path(r'^/vc/.*', '/var/www')
+        # Sets MY_PATH to be the path of the installed agent.
+        MY_PATH = os.path.dirname(__file__)
+        # Sets WEBROOT to be the path to the webroot directory
+        # in the agent-data directory of the installed agent.
+        WEBROOT = os.path.join(MY_PATH, "webroot")
+        # Serves the static content from 'webroot' directory
+        self.vip.web.register_path("/simpleweb", WEBROOT)
```

Comment:
Does this allow relative imports...eg /simpleweb/image/534.jpg?  If not please modify like the vc regular expression that was removed.